src/gui_common/ChemicalEquation.cs

```diff
@@ -52,7 +52,7 @@ public class ChemicalEquation : VBoxContainer
             if (equationFromProcess == null && value == null)
                 return;
 
-            if (equationFromProcess?.Equals(value) == true)
+            if (value != null && equationFromProcess?.Equals(value) == true)
```

Comment:
Why null check here?

Reply:
The Equals method here takes in a non-nullable parameter so it causes a warning because `value` is nullable here. I think this is semantically the right way to express what the old code wanted to do.